link.x

```diff
@@ -54,7 +54,10 @@ SECTIONS
     *(.bss .bss.*);
     . = ALIGN(4);
     _ebss = .;
-  } > RAM
+  } > RAM AT > FLASH
+  /* NOTE(AT > FLASH) without this LLD v6 produces a binary that crashes OpenOCD whereas LLD v7
+     emits a ".rodata and .bss sections overlap" error ... This hacky workaround doesn't increase
+     the binary size AFAICT */
```

Comment:
One drawback is this will make firmwares with *really* full flashes uncompilable. Do I understand it right that lld v7 doesn't require the workaround?

Reply:
> Do I understand it right that lld v7 doesn't require the workaround?

No, this also needed for the v7 I tested. Without the workaround linking with LLD fails with the error message in the comment; GNU LD can link the same program without errors and produces a binary that doesn't crash OpenOCD.

Reply:
Are you going to look at the underlying lld bug? I don't yet use lld for my embedded development so I'm afraid you're in the better position here. I can review though.

Reply:
I will at least submit a bug report. I have zero experience hacking LLVM :-).